File: backend/certbot/request/get_domains.py
from dataclasses import dataclass, field

from api.base_request import BaseRequest
from api.base_response import BaseResponse
from certbot.data.domain import Domain


@dataclass
class GetDomainsRequest(BaseRequest):
    pass


@dataclass
class GetDomainsResponse(BaseResponse):
    domains: list[Domain] = field(default_factory=list)

File: frontend/src/api/certbot/request/get_domains.ts
import { api } from 'boot/axios';
import API from 'api/certbot/api';
import Domain from 'api/certbot/data/domain';

export interface IGetDomainsResponse extends Response {
  domains: Domain[];
}

export default function getDomains() {
  return api.get<IGetDomainsResponse>(API.DOMAINS);
}
